### DISCUSSION TITLE
App Directory how can I listen route events ?

### DISCUSSION THREAD

**Discussion by hungcrush** [vercel/next.js#44799] [Help] [answered]
I'm using NextJS 13 and appdir. Seems routes events is not implemented yet ? currently what's the best way to listen ?

**Comment by icyJoseph** [comment, marked as answer]
Please refer to: https://github.com/vercel/next.js/discussions/41745#discussioncomment-3986452

In general, the problem is that the routing is now fully integrated with React's internals, which means that routing is concurrent, and the idea of events is slightly more difficult to implement. Previously answered in: https://github.com/vercel/next.js/discussions/43950

Perhaps also relevant: https://github.com/vercel/next.js/discussions/41745#discussioncomment-4360144

**Reply by hungcrush** [reply]
Thank you so much
